misk-testing/src/main/kotlin/misk/testing/ExternalDependency.kt

```diff
@@ -0,0 +1,34 @@
+package misk.testing
+
+/**
+ * An external dependency of the Misk Application that needs to be started for test, like Redis,
+ * Vitess, MySQL, SQS, etc.
+ */
+interface ExternalDependency {
+  /**
+   * Starts the dependency.
+   */
+  fun startup()
+
+  /**
+   * Stops the dependency.
+   */
+  fun shutdown()
+
+  /**
+   * Called before each test run
+   */
+  fun beforeEach() {}
```

Comment:
I'd prefer we just force impls to extend this so they have to think about it.

Reply:
This is a default function. Impls do need to extend it